esi/lib/node.py

```diff
@@ -16,93 +17,125 @@
 
 
 def network_list(connection, filter_node=None, filter_network=None):
-    """List nodes and their network attributes
-
-    :param connection: An OpenStack connection
-    :type connection: :class:`~openstack.connection.Connection`
-    :param node: Name or UUID of the node
-    :param network: The name or ID of a network
-    """
 
-    if filter_node:
-        machine = connection.baremetal.find_node(filter_node, ignore_missing=False)
-        ports = connection.list_nics_for_machine(uuid=machine.id)
-
-        node_name = machine.name
-    else:
-        ports = None
-        nodes = None
+    def get_baremetal_info(filter_node):
```

Comment:
I don't see a strong reason to keep this function inside `network_list`; it's potentially something that someone else might want to call. I would rename it to something like `node_and_port_list`. The reason is that node and port are Ironic objects, while baremetal refers to the entire service.

Also, I'd use `filter_node=None` here